sandbox/lua/decoders/docker_fluentd.lua

```diff
@@ -0,0 +1,113 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+Decode data from Docker's Fluentd logging driver.
+
+**Note**: The Fluentd logging driver is available in Docker 1.8.0rc1 and later.
+
+Config:
+
+- type (string, optional, default nil):
+    Sets the message 'Type' header to the specified value
+
+*Example Heka Configuration*
+
+.. code-block:: ini
+
+    [FluentdInput]
+    type = "TcpInput"
+    address = ":24224"
+    splitter = "MessagePackSplitter"
+    decoder = "DockerFluentdDecoder"
+
+    [MessagePackSplitter]
+    # No config
+
+    [DockerFluentdDecoder]
+    type = "SandboxDecoder"
+    filename = "lua_decoders/docker_fluentd.lua"
+
+        [DockerFluentdDecoder.config]
+        type = "docker-fluentd"
+
+*Example Heka Message*
+
+.. code-block:: bash
+
+    docker run \
+        --log-driver fluentd \
+        --log-opt fluentd-address=YOUR_HEKA:24224 \
+        -d busybox \
+        echo Hello world
+
+This command should generate something like this:
+
+:Timestamp: 2015-08-03 20:41:06 +0000 UTC
+:Type: docker-fluentd
+:Hostname: 192.168.59.103:60088
+:Pid: 0
+:Uuid: da45d947-037d-4870-abbe-671d820ebe8d
+:Logger: stdout
+:Payload: Hello world
+:EnvVersion: 
+:Severity: 7
+:Fields:
+    | name:"container_name" type:string value:"/suspicious_meitner"
+    | name:"tag" type:string value:"docker.7dc19982364b"
+    | name:"container_id" type:string value:"7dc19982364ba459958041d2fe85e8bdc3825d06397296ddd981c51e5f15cb89"
+
+--]]
+
+local mp = require "msgpack"
+
+local msg_type = read_config("type")
+
+local msg = {
+    Timestamp  = nil,
+    EnvVersion = nil,
+--    Hostname   = nil,
```

Comment:
We should probably keep this value in the table and explicitly propagate the incoming message's Hostname value since in case the input has already set it to a useful value by the time splitting happens.